src/node_crypto.cc

```diff
@@ -1795,6 +1795,25 @@ static bool SafeX509ExtPrint(BIO* out, X509_EXTENSION* ext) {
 }
 
 
+static void addFingerprintDigest(unsigned char* md,
+                                 unsigned int md_size,
+                                 char* fingerprint) {
```

Comment:
If you change this to `char (*fingerprint)[3 * EVP_MAX_MD_SIZE + 1]`, then people can never inadvertently pass a buffer that's too small and the compiler is more likely to warn you when you try to write past the end.  You index into it with `(*fingerprint)[3*i] = ...`.